alua.c

```diff
@@ -483,10 +519,28 @@ int tcmu_transition_tgt_port_grp(struct tgt_port_grp *group, uint8_t new_state,
 	return SAM_STAT_GOOD;
 }
 
+static int tcmu_report_state(struct tcmu_device *dev,
+			     struct tgt_port_grp *group)
+{
+	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
+	int ret;
+
+	/* TODO: For ESX return remote ports */
+
+	if (!rhandler->has_lock)
+		return group->state;
+
+	ret = rhandler->has_lock(dev);
+	if (ret <= 0) {
+		return ALUA_ACCESS_STATE_STANDBY;
+	} else {
+		return ALUA_ACCESS_STATE_OPTIMIZED;
+	}
+}
+
 int tcmu_emulate_report_tgt_port_grps(struct tcmu_device *dev,
 				      struct list_head *group_list,
-				      struct tcmulib_cmd *cmd,
-				      tcmu_report_state_fn_t *report_fn)
+				      struct tcmulib_cmd *cmd)
```

Comment:
You saw where I said I am fine with making it runner specific, right? Are you also not going to modify the cmake so it is built as part of tcmu-runner instead of libtcmu?

It is fine to start small and build on it, but if everyone has their own policy about where to add fields and build files we end up with a mix match and someone has to clean it up eventually.

If we are saying that there is no libtcmu user that implements its own daemon and there probably will not be, then lets discuss that in another PR. Trying to maintain that design is a pain, but I was under the impression at some point there were outside users, because they did some of the low level libtcmu work.